meson.build

```diff
@@ -0,0 +1,118 @@
+#
+# - major version is automatically generated from VA-API major version
+# - minor version is automatically generated from VA-API minor version
+# - increment micro for any library release
+# - reset micro version to zero when VA-API major or minor version is changed
+project(
+  'libva', 'c',
+  version : '2.0.1.1',
+  meson_version : '>= 0.37.0',
+  default_options : [ 'warning_level=1',
+                      'buildtype=debugoptimized' ])
+
+# VA-API version
+# - increment major for any ABI change
+# - increment minor for any interface change (e.g. new/modified function)
+# - increment micro for any other change (new flag, new codec definition, etc.)
+# - reset micro version to zero when minor version is incremented
+# - reset minor version to zero when major version is incremented
+va_api_major_version = 1
+va_api_minor_version = 0
+va_api_micro_version = 0
+
+va_api_version = '@0@.@1@.@2@'.format(va_api_major_version,
+				      va_api_minor_version,
+				      va_api_micro_version)
+
+version_arr = meson.project_version().split('.')
+libva_major_version = version_arr[0]
+libva_minor_version = version_arr[1]
+libva_micro_version = version_arr[2]
+libva_version = '@0@.@1@.@2@'.format(libva_major_version,
+				     libva_minor_version,
+				     libva_micro_version)
+if version_arr.length() == 4
+  libva_version = '@0@.pre@1@'.format(libva_version, version_arr[3])
+endif
+
+
+# libva library version number (generated, do not change)
+# XXX: we want the SONAME to remain at libva.so.1 for VA-API major == 0
+#
+# The library name is generated libva.<x>.<y>.0 where
+# <x> = VA-API major version + 1
+# <y> = 100 * VA-API minor version + VA-API micro version
+#
+# For example:
+# VA-API 0.32.0 generates libva.so.1.3200.0
+# VA-API 0.34.1 generates libva.so.1.3401.0
+# VA-API 1.2.13 generates libva.so.2.213.0
+libva_interface_bias = va_api_major_version + 1
+libva_interface_age = 0
+libva_binary_age = 100 * va_api_minor_version + va_api_micro_version - libva_interface_age
+
+libva_lt_current = 100 * va_api_minor_version + va_api_micro_version + libva_interface_bias
+libva_lt_revision = libva_interface_age
+libva_lt_age = libva_binary_age - libva_interface_age
+
+libva_lt_version = '@0@.@1@.@2@'.format(libva_lt_current,
+					libva_lt_revision,
+					libva_lt_age)
+
+driverdir = get_option('driverdir')
+if driverdir == ''
+  driverdir = '@0@/@1@/@2@'.format(get_option('prefix'), get_option('libdir'), 'dri')
+endif
+
+configinc = include_directories('.')
+
+cc = meson.get_compiler('c')
+dl_dep = cc.find_library('dl', required : false)
+
+libdrm_dep = dependency('libdrm', version : '>= 2.4')
+
+if get_option('with_x11') != 'no' or get_option('with_glx') != 'no'
+  glx_is_required = (get_option('with_x11') == 'yes'
+		     or get_option('with_glx') != 'yes')
+
+  x11_dep = dependency('x11', required : glx_is_required)
+  xext_dep = dependency('xext', required : glx_is_required)
+  xfixes_dep = dependency('xfixes', required : glx_is_required)
+endif
```

Comment:
The following cases do not work here:

1. with_x11=='no', with_glx=='auto' should disable x11 and glx
2. with_x11=='no', with_glx=='yes' should generate configure error (i.e. combination not allowed since glx requires x11).

Please refer to configure.ac for example of how autoconf handles this.

Reply:
> Please refer to configure.ac for example of how autoconf handles this.

At the end I finished with a logical copy of configure.ac, though I wanted to simplify it. But it seems the correct way. Thanks.